src/Compilers/CSharp/Portable/BoundTree/BoundExpressionExtensions.cs

```diff
@@ -247,10 +247,16 @@ internal static TypeSymbolWithAnnotations GetTypeAndNullability(this BoundExpres
                         var conversion = (BoundConversion)expr;
                         if (conversion.ExplicitCastInCode)
                         {
-                            return conversion.IsExplicitlyNullable;
+                            if (conversion.IsExplicitlyNullable)
+                            {
+                                return true;
+                            }
+                            else if (expr.ConstantValue == null)
+                            {
+                                return false;
```

Comment:
>return false; [](start = 32, length = 13)

It doesn't feel like this is applicable in general case. If conversion doesn't have a constant value, it doesn't mean the result is not nullable. It is probably more appropriate to specially handle conversions for parameter default values because non-constant cases for reference types are errors anyway.

Reply:
Also it is not obvious how this change makes a difference. Could you provide some details, please.

---
In reply to: [187171161](https://github.com/dotnet/roslyn/pull/26726#discussion_r187171161) [](ancestors = 187171161)